samples/geometry/Sample_VertexAnimation.ts

```diff
@@ -0,0 +1,68 @@
+import { View3D, PlaneGeometry, Engine3D, Scene3D, AtmosphericComponent, CameraUtil, HoverCameraController, Object3D, DirectLight, KelvinUtil, MeshRenderer, LitMaterial, VertexAttributeName, Time } from "@orillusion/core";
+import { Stats } from "@orillusion/stats";
+import { GUIUtil } from "@samples/utils/GUIUtil";
+
+// An sample of dynamically updating a geometry vertex attribute
+class Smaple_VertexAnimation {
+    // This geometry will dynamically update its vertex data over time
+    floorGeometry: PlaneGeometry;
+    scene: Scene3D;
+    async run() {
+        await Engine3D.init({ beforeRender: () => this.update() });
+
+        let view = new View3D();
+
+        view.scene = new Scene3D();
+        view.scene.addComponent(AtmosphericComponent);
+        view.scene.addComponent(Stats);
+
+        this.scene = view.scene;
+
+        view.camera = CameraUtil.createCamera3DObject(view.scene, "camera");
+        view.camera.perspective(60, Engine3D.aspect, 1, 2000);
+        view.camera.object3D.addComponent(HoverCameraController).setCamera(35, -20, 150);
+
+        Engine3D.startRenderView(view);
+
+        this.createScene();
+    }
+
+    private createScene() {
+        // add light
+        let lightObj3D = new Object3D();
+        let directLight = lightObj3D.addComponent(DirectLight);
+        directLight.intensity = 25;
+        directLight.lightColor = KelvinUtil.color_temperature_to_rgb(5355);
+        directLight.castShadow = true;
+        lightObj3D.rotationX = 53.2;
+        lightObj3D.rotationY = 220;
+        lightObj3D.rotationZ = 5.58;
+        GUIUtil.renderDirLight(directLight);
+
+        this.scene.addChild(lightObj3D);
+
+        // add floor
+        this.floorGeometry = new PlaneGeometry(100, 100, 199, 199);
+
+        let floor = new Object3D();
+        let renderer = floor.addComponent(MeshRenderer);
+        renderer.geometry = this.floorGeometry;
+        renderer.material = new LitMaterial();
+        this.scene.addChild(floor);
+    }
+
+    private update() {
+        // update its vertex data over time
+        if (this.floorGeometry) {
+            let attribute = this.floorGeometry.getAttribute(VertexAttributeName.position);
```

Comment:
It's best to do water wave calculations here

Reply:
Please see line 60.
It's already look like a wave.

Reply:
![image](https://user-images.githubusercontent.com/10559053/236983277-b2f035b4-7b70-4e11-83b6-790c19b1a773.png)

Reply:
![image](https://github.com/Orillusion/orillusion/assets/10559053/1d6a2964-1f73-4265-8bf9-b8f734fb534d)

after computeNormals